Indigenous NT 4th Request

ind-nt opened the pull request.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

DavidLRowe commented:
NB: This is a resubmission of PR #803.

In the indigenous_nt folder:
- delete the indigenous_NT.keyboard_info file (but keep the indigenous_nt.keyboard_info file!)
- delete the indigenous_nt.kps file (but leave the one in the source folder)
- restore the indigenous_nt.kpj (Keyman project) file

The copyright statement, which appears in numerous files, needs to have a name (either of a person or a legally recognised organisation) as well as the date (or date range). All these statements must agree. (Right now the indigenous_nt.kmn file has "2013-2019", whereas the README.md has "2019".) You'll need to check/change:
- README.md
- LICENSE.md
- indigenous_nt.kmn
- HISTORY.md (if you want to include information supporting the 2013 date)
- indigenous_nt.kps (Keyman package source)

In the indigenous_nt.keyboard_info file, some of the language tags will need a script modifier and some should not have one. The information at 
https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry will indicate with "Suppress-Script: Latn" that the script name should be dropped. For the ones you have listed, only "en" and "fr" should drop the script. All the others should have "-Latn". For "en" and "fr" you may want to add the region "-CA".

Our preference is that each language tag be listed on a separate line in the indigenous_nt.keyboard_info file, for example:
```
"languages": [
    "en-CA",
    "fr-CA",
    "ikt-Latn",
    "esi-Latn",
    "chp-Latn",
    "cr-Latn",
    "dgr-Latn",
    "gwi-Latn",
    "xsl-Latn",
    "scs-Latn"],
```

The list of languages in the package source (indigenous_nt.kps) file should be the same as the list of languages in the indigenous_nt.keyboard_info file, and preferably in the same order.

For the touch layout, you currently have phone and tablet layouts, but they are identical. Unless you plan to make them different, it is better to just have one, so that you don't have to update both layouts, should any changes be needed. Keyman will use the one existing layout for both platforms. 

Because you are using Shift+Alt+O and Shift+Alt+M, you need to warn users that they will need to change the default keys that Keyman uses. The following text (or something like it) should be included in the welcome.htm and the indigenous_nt.php files.
```
This keyboard defines some keys that are set by default as "hotkeys" when Keyman is installed. 
To use this keyboard, you will want to go to <b>Configuration.../Hotkeys</b>, 
select the hotkey name, and either change the hotkey or turn it off.  Currently 
"Switch Keyman Desktop Off" uses Shift+Alt+O
"Open Keyboard Menu" uses Shift+Alt+M
```

I'm wondering about the choice of the "dotless i" as the base character for "i" on the touch layout. Most fonts handle removing the dot when displaying a diacritic. Do you need to type a dotless i very often, if at all? If not, consider moving the regular "i" to the base character and putting the dotless i as a longpress on that key.

DavidLRowe commented:
A minor point on the touch layout: Do you need the "L with stroke" as a longpress key on "L" when it has its own key right next to the "L"? This applies to the default and the shift layers.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

ind-nt commented:
> 
> 
> A minor point on the touch layout: Do you need the "L with stroke" as a longpress key on "L" when it has its own key right next to the "L"? This applies to the default and the shift layers.

Removed.

ind-nt commented:
> 
> 
> NB: This is a resubmission of PR #803.
> 
> In the indigenous_nt folder:
> 
>     * delete the indigenous_NT.keyboard_info file (but keep the indigenous_nt.keyboard_info file!)
> 
>     * delete the indigenous_nt.kps file (but leave the one in the source folder)
> 
>     * restore the indigenous_nt.kpj (Keyman project) file
> 
> 
> The copyright statement, which appears in numerous files, needs to have a name (either of a person or a legally recognised organisation) as well as the date (or date range). All these statements must agree. (Right now the indigenous_nt.kmn file has "2013-2019", whereas the README.md has "2019".) You'll need to check/change:
> 
>     * README.md
> 
>     * LICENSE.md
> 
>     * indigenous_nt.kmn
> 
>     * HISTORY.md (if you want to include information supporting the 2013 date)
> 
>     * indigenous_nt.kps (Keyman package source)
> 
> 
> In the indigenous_nt.keyboard_info file, some of the language tags will need a script modifier and some should not have one. The information at
> https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry will indicate with "Suppress-Script: Latn" that the script name should be dropped. For the ones you have listed, only "en" and "fr" should drop the script. All the others should have "-Latn". For "en" and "fr" you may want to add the region "-CA".
> 
> Our preference is that each language tag be listed on a separate line in the indigenous_nt.keyboard_info file, for example:
> 
> ```
> "languages": [
>     "en-CA",
>     "fr-CA",
>     "ikt-Latn",
>     "esi-Latn",
>     "chp-Latn",
>     "cr-Latn",
>     "dgr-Latn",
>     "gwi-Latn",
>     "xsl-Latn",
>     "scs-Latn"],
> ```
> 
> The list of languages in the package source (indigenous_nt.kps) file should be the same as the list of languages in the indigenous_nt.keyboard_info file, and preferably in the same order.
> 
> For the touch layout, you currently have phone and tablet layouts, but they are identical. Unless you plan to make them different, it is better to just have one, so that you don't have to update both layouts, should any changes be needed. Keyman will use the one existing layout for both platforms.
> 
> Because you are using Shift+Alt+O and Shift+Alt+M, you need to warn users that they will need to change the default keys that Keyman uses. The following text (or something like it) should be included in the welcome.htm and the indigenous_nt.php files.
> 
> ```
> This keyboard defines some keys that are set by default as "hotkeys" when Keyman is installed. 
> To use this keyboard, you will want to go to <b>Configuration.../Hotkeys</b>, 
> select the hotkey name, and either change the hotkey or turn it off.  Currently 
> "Switch Keyman Desktop Off" uses Shift+Alt+O
> "Open Keyboard Menu" uses Shift+Alt+M
> ```
> 
> I'm wondering about the choice of the "dotless i" as the base character for "i" on the touch layout. Most fonts handle removing the dot when displaying a diacritic. Do you need to type a dotless i very often, if at all? If not, consider moving the regular "i" to the base character and putting the dotless i as a longpress on that key.

I believe I've worked through all the issues outlined above. I have retained the dotless i which is standard with most of the Dene languages (except Gwich'in). Often its presence shows that active attention is being paid to proper transcription.

LornaSIL commented:
There are a few issues that still need resolving:
* The welcome file is not showing the graphic. Were you wanting the graphic to display? If so, please add a link to the graphic.
* Please add this line to the .kmn source:
```store(&VERSION) "10.0"```
* The .kps file uses the wrong path for most of the files. In the end I created a whole new .kps file before I could get the keyboard to build. I'll attach it here in case it's easier for you to use this file. 
* arcticfox.png was removed from your repo, but it is still referenced in the .kps file. I removed it from the .kps file that I'm attaching. If you want the articifox.png file you should add it back to the repo and then add it back to the .kps file.
* Reopen the .kpj file so it finds the correct paths.

I hope this is straightforward.

[indigenous_nt.zip](https://github.com/keymanapp/keyboards/files/3570484/indigenous_nt.zip)

keyman-server commented:
This pull request is from an external repo and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

ind-nt commented:
Thanks Lorna. All that you suggested has been corrected.

ind-nt commented:
Thanks Lorna, I believe I’ve fixed all the items you cited below.

- R

On Tue, Sep 3, 2019 at 10:42 AM Lorna Evans <notifications@github.com>
wrote:

> There are a few issues that still need resolving:
>
>    - The welcome file is not showing the graphic. Were you wanting the
>    graphic to display? If so, please add a link to the graphic.
>    - Please add this line to the .kmn source:
>    store(&VERSION) "10.0"
>    - The .kps file uses the wrong path for most of the files. In the end
>    I created a whole new .kps file before I could get the keyboard to build.
>    I'll attach it here in case it's easier for you to use this file.
>    - arcticfox.png was removed from your repo, but it is still referenced
>    in the .kps file. I removed it from the .kps file that I'm attaching. If
>    you want the articifox.png file you should add it back to the repo and then
>    add it back to the .kps file.
>    - Reopen the .kpj file so it finds the correct paths.
>
> I hope this is straightforward.
>
> indigenous_nt.zip
> <https://github.com/keymanapp/keyboards/files/3570484/indigenous_nt.zip>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/keymanapp/keyboards/pull/815?email_source=notifications&email_token=AMTEXA25CEUZQRFXHRMSSILQH2HYTA5CNFSM4IEVK3I2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5Y2E5Y#issuecomment-527540855>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMTEXA6FKUMDQXNSYNKVSNDQH2HYTANCNFSM4IEVK3IQ>
> .
>
-- 
________________________________________
Rajiv Rawat, Yellowknife, NT, Canada
C: 1-867-446-7267

LornaSIL commented:
The build system says there is a problem with the keyboard_info file. I've run it through a json validator which says it's valid. So, at this point I only know how to test by asking you to change a few things and try again.
Please remove these lines and commit the file again:
```  "sourcePath": "release/i/indigenous_nt",
  "version": "3.2",
  "packageFilename": "indigenous_nt.kmp",
  "encodings": [
    "unicode"
  ],
  "packageFileSize": 114526,
  "packageIncludes": [
    "visualKeyboard",
    "documentation",
    "welcome"
  ],
  "minKeymanVersion": "9.0",
```
None of them are really necessary, the home page will show details automatically without those lines and the minKeymanVersion is definitely wrong. It would be 10.0. So, try removing those lines and I'll try and build again.

ind-nt commented:
Ok done.

Thanks!

On Tue, Sep 3, 2019 at 12:18 PM Lorna Evans <notifications@github.com>
wrote:

> The build system says there is a problem with the keyboard_info file. I've
> run it through a json validator which says it's valid. So, at this point I
> only know how to test by asking you to change a few things and try again.
> Please remove these lines and commit the file again:
>
>   "version": "3.2",
>   "packageFilename": "indigenous_nt.kmp",
>   "encodings": [
>     "unicode"
>   ],
>   "packageFileSize": 114526,
>   "packageIncludes": [
>     "visualKeyboard",
>     "documentation",
>     "welcome"
>   ],
>   "minKeymanVersion": "9.0",
>
> None of them are really necessary, the home page will show details
> automatically without those lines and the minKeymanVersion is definitely
> wrong. It would be 10.0. So, try removing those lines and I'll try and
> build again.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/keymanapp/keyboards/pull/815?email_source=notifications&email_token=AMTEXA2DC4CHI2PZZKUZBU3QH2S53A5CNFSM4IEVK3I2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5ZDC3Y#issuecomment-527577455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMTEXAZ4TP73VANQ67AAACLQH2S53ANCNFSM4IEVK3IQ>
> .
>


-- 
________________________________________
Rajiv Rawat, Yellowknife, NT, Canada
C: 1-867-446-7267

LornaSIL commented:
@mcdurdin I can't see any obvious problem with the keyboard_info file. The last build I triggered said "Failed to validate indigenous_nt.keyboard_info in /c/BuildAgent/work/aee9ab97cd908689/keyboards/release/i/indigenous_nt/"

I ran it through a json validator and it's okay. So, there's some syntax I'm not used to that is wrong.
Are we allowed to use `\u02BC` format in the keyboard_info?

```
{
  "license": "mit",
  "description": "This is the Keyman edition of an universal keyboard for the official languages of the Northwest Territories. It retains the standard US keyboard layout while providing access to special characters and diacritics through the right Alt key. The mobile version allows multiple character forms through the longpress popup method.",
  "id": "indigenous_nt",
  "languages": {
    "en-CA": {
      "displayName": "English",
      "languageName": "English"
    },
    "fr-CA": {
      "displayName": "French",
      "languageName": "French"
    },
    "ikt-Latn": {
      "displayName": "Inuinnaqtun",
      "languageName": "Inuinnaqtun"
    },
    "esi-Latn": {
      "displayName": "Inuvialuktun",
      "languageName": "Inuvialuktun"
    },
    "cr-Latn": {
      "displayName": "Cree",
      "languageName": "Cree"
    },
    "chp-Latn": {
      "displayName": "Chipewyan",
      "languageName": "Chipewyan"
    },
    "dgr-Latn": {
      "displayName": "Dogrib",
      "languageName": "Dogrib"
    },
    "gwi-Latn": {
      "displayName": "Gwich\u02BCin",
      "languageName": "Gwich\u02BCin"
    },
    "xsl-Latn": {
      "displayName": "South Slavey",
      "languageName": "South Slavey"
    },
    "scs-Latn": {
      "displayName": "North Slavey",
      "languageName": "North Slavey"
    }
  },
  "name": "Indigenous NT Integrated Keyboard",
  "authorName": "Rajiv Rawat",
  "authorEmail": "rajiv_rawat@gov.nt.ca",
  "lastModifiedDate": "2019-09-03T15:53:07.007Z",
  "sourcePath": "release/i/indigenous_nt",
  "helpLink": "https://help.keyman.com/keyboard/indigenous_nt",
  "platformSupport": {
    "windows": "full",
    "macos": "full",
    "linux": "full",
    "desktopWeb": "full",
    "mobileWeb": "full",
    "android": "full",
    "ios": "full"
  }
}
```

mcdurdin commented:
While the JSON is valid structurally, it doesn't meet the requirements for a .keyboard_info file: the `languages` key has invalid entries in it -- we don't support `displayName` and `languageName`. It should probably be just an array of BCP 47 strings (e.g. `['en','fr']`).

FWIW, you can also get more detailed validation errors by visiting https://jsonschemalint.com/#/version/draft-06/markup/json and pasting the source schema (from https://api.keyman.com/schemas/keyboard_info.source.json) into the schema field.

Note: you will need to delete the line `"$schema": "http://json-schema.org/schema#",` from the schema itself, before the schema itself validates with that tool).

You can then paste the .keyboard_info into the right hand box and get more detailed results, e.g.:
![image](https://user-images.githubusercontent.com/4498365/64215739-5b923d80-cef9-11e9-9343-4e3844d7fd4e.png)

LornaSIL commented:
@ind-nt can you delete the display name and language name. Earlier in this thread it shows the correct syntax.
Hopefully this will make it build!

ind-nt commented:
Ok done.. thanks for all your help!

On Tue, Sep 3, 2019 at 6:14 PM Lorna Evans <notifications@github.com> wrote:

> @ind-nt <https://github.com/ind-nt> can you delete the display name and
> language name. Earlier in this thread it shows the correct syntax.
> Hopefully this will make it build!
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/keymanapp/keyboards/pull/815?email_source=notifications&email_token=AMTEXA6UI5S4G6MMWJRK46DQH34U3A5CNFSM4IEVK3I2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5Z57JQ#issuecomment-527687590>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMTEXA7DFLP2ZR4RUTFPFHLQH34U3ANCNFSM4IEVK3IQ>
> .
>
-- 
________________________________________
Rajiv Rawat, Yellowknife, NT, Canada
C: 1-867-446-7267

LornaSIL commented:
I'm sorry this is such a headache. The keyboard_info file is still not compiling. Try the attached one. I've added back in the scripts and deleted some of the extra information that is not needed. Much of it is automatically created from other places. I left you as author and your email address as that is not pulled in from elsewhere.

[indigenous_nt.zip](https://github.com/keymanapp/keyboards/files/3575017/indigenous_nt.zip)

ind-nt commented:
Hi Lorna,

Ok I've uploaded it.

Yeah, it's pretty finicky. Maybe the help pages on the website should have
snapshots of the proper file organization. The team is pretty awesome
though!

- R

On Wed, Sep 4, 2019 at 8:41 AM Lorna Evans <notifications@github.com> wrote:

> I'm sorry this is such a headache. The keyboard_info file is still not
> compiling. Try the attached one. I've added back in the scripts and deleted
> some of the extra information that is not needed. Much of it is
> automatically created from other places. I left you as author and your
> email address as that is not pulled in from elsewhere.
>
> indigenous_nt.zip
> <https://github.com/keymanapp/keyboards/files/3575017/indigenous_nt.zip>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keymanapp/keyboards/pull/815?email_source=notifications&email_token=AMTEXAZHBURT75TI5UW5OA3QH7CLFA5CNFSM4IEVK3I2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD53ZY3Q#issuecomment-527932526>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMTEXA5O54JEJR44QA7EQJ3QH7CLFANCNFSM4IEVK3IQ>
> .
>


-- 
________________________________________
Rajiv Rawat, Yellowknife, NT, Canada
C: 1-867-446-7267

LornaSIL commented:
This keyboard has built and is available here: https://keyman.com/keyboards/indigenous_nt
Unfortunately the help (php) file is referencing a file named ```indigenous_nt_layout.pdf```, but the name in the repo is ```indigenous_nt-layout.pdf``` so people can't download the pdf.

Could you submit a new PR changing the name of the file from ```indigenous_nt-layout.pdf``` to ```indigenous_nt_layout.pdf```

This is the easiest solution and will not require a version bump to the keyboard.

ind-nt commented:
Ok done.  Sorry about that.

On Wed, Sep 4, 2019 at 1:50 PM Lorna Evans <notifications@github.com> wrote:

> This keyboard has built and is available here:
> https://keyman.com/keyboards/indigenous_nt
> Unfortunately the help (php) file is referencing a file named
> indigenous_nt_layout.pdf, but the name in the repo is
> indigenous_nt-layout.pdf so people can't download the pdf.
>
> Could you submit a new PR changing the name of the file from
> indigenous_nt-layout.pdf to indigenous_nt_layout.pdf
>
> This is the easiest solution and will not require a version bump to the
> keyboard.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keymanapp/keyboards/pull/815?email_source=notifications&email_token=AMTEXAZXLFUOC3VO5VGTAW3QIAGPNA5CNFSM4IEVK3I2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD54YLPI#issuecomment-528057789>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMTEXA7QRZG6X5FMHMIVJXDQIAGPNANCNFSM4IEVK3IQ>
> .
>
-- 
________________________________________
Rajiv Rawat, Yellowknife, NT, Canada
C: 1-867-446-7267
